### ISSUE TITLE
Is it possible to approve a pull request from within forge?

### ISSUE THREAD

**Issue by BigTheta** [magit/forge#377]
I have checked the manual, FAQ and reddit, I don't see anywhere where its even been asked before. 

I was asked to review a pull request, I was able to view the commits and comment on the PR. Now, I would like to approve the PR. I know `gh` has this functionality. It seems like the sort of functionality forge was intended to support?

**Comment by tarsius** [comment]
Not yet, I believe. Supporting it isn't a top priority but eventually it will probably be implemented. (The projects I am involved in for the most part work by merging or not merging pull-requests without explicitly and beforehand approve or reject it.)

**Comment by jpneverwas** [comment]
Thanks for making this package. It's truly amazing.

I'm not sure whether this is related. If not, I can open a new issue. My problem is that I'm unable to see the second post in [this thread](https://github.com/dustin/go-nntp/pull/11). At the time of writing, there are three posts. The first and last show up, but the one in the middle, the reply from the repo's owner, is missing. I figured maybe it's related because the green checkmark icon and surrounding UI seem to indicate it's not a normal post (sorry if this is common knowledge, I'm a GitHub dunce). I checked the database manually with the command line tool, and the post is absent there as well. As you can see, I  punted and just replied via email like a schmuck.

Thanks again for all your hard work. Apologies if something obvious is going on here and I've just wasted your time.

**Comment by tarsius** [comment]
> Is it possible to join/approve a pull request from within forge?

I just remembered that I implemented this quite recently in 3112aded28ad8d17d30980168d6340b6391f0d14. It's available from the merge menu, but not by default so...

**Comment by tarsius** [comment]
Certain transient popups hide some of their suffix commands by default. These additional suffixes can be [enabled interactively](https://magit.vc/manual/transient/Enabling-and-Disabling-Suffixes.html). TL;DR Enter the transient as usual and type <kbd>C-x l</kbd>.

<sub>This is a [canned response](https://help.github.com/en/github/writing-on-github/working-with-saved-replies); it may not apply 100% in this case.</sub>

**Comment by tarsius** [comment]
> I'm not sure whether this is related.

It's completely unrelated.

> If not, I can open a new issue.

See #75 instead.

**Comment by tarsius** [comment]
> I have checked the manual,

I've just added it to the manual.

**Comment by BigTheta** [comment]
Thank you. Much appreciated.

**Comment by BigTheta** [comment]
Sorry, I never actually attempted this until just now.

> I just remembered that I implemented this quite recently in [3112ade]

To be clear, you are taking about merging (not approving?) I don't see anything in that commit regarding approving reviews.

> It's available from the merge menu, but not by default so...

I'm looking at the merge menu with suffixes enabled and I still don't see anything regarding approving the topic?

![image](https://user-images.githubusercontent.com/825406/131706446-febbc974-2667-4e2b-98ec-24bd0817c35f.png)

**Comment by tarsius** [comment]
> To be clear, you are taking about merging (not approving?)

Yeah. Your use of the word "join" confused me. That sounded like a synonym for "merge".

So no. Improving still isn't supported and adding support is a low priority.

**Comment by aiguofer** [comment]
Check out https://github.com/wandersoncferreira/code-review (I just found it looking for better ways to do reviews using forge)

Slightly weird switching between `forge` and `code-review`, but they do integrate fairly well.

**Comment by magandrez** [comment]
@BigTheta If you are still interested on this, you can check out https://github.com/charignon/github-review

**Comment by tarsius** [comment]
I actually implemented that last week but wanted to look into #313 before pushing anything.  Trying to implement #313 unfortunately put me into a refactor frenzy, so I had to abandon that, to refocus on other more straight forward parts of the backlog.

The new commands on `/A` and `/R` in `forge-topic-menu` (`<C-return>`) let you write a message (which is optional you can just leave the buffer empty), but that's it.  There's no support for commenting on specific parts of the pull-request.  #75 is the issue about that.

**Comment by vunhan** [comment]
Could we have this feature support for Gitlab too?

**Comment by tarsius** [comment]
> Could we have this feature support for Gitlab too?

I plan to start focusing on GitLab in about two months. Until then I intend to focus on other parts of the backlog, even when it means that GitLab support falls behind some more.

A big ticket when it comes to improved GitLab support is to switch to their GraphQL API, for those features (mostly queries but not mutations) that already use GraphQL in the GitHub backend.

Many features that are currently missing in the GitLab backend would require some REST now, and I avoid doing that because I would soon have to redo that work using GraphQL.
